app/src/main/java/com/mapbox/vision/examples/models/UiSign.kt

```diff
@@ -2,58 +2,63 @@ package com.mapbox.vision.examples.models
 
 import com.mapbox.vision.mobile.core.models.classification.FrameSignClassifications
 
-data class UiSign(val signType: SignType, val signNum: SignNumber) {
+sealed class UiSign(val signType: SignType) {
 
     companion object {
         fun getUiSigns(frameSignClassifications: FrameSignClassifications): List<UiSign> {
             val signTypeValues = SignType.values()
 
             return frameSignClassifications.signs.map { signClassification ->
                 val signTypeIndex = signClassification.sign.type.ordinal
-                if (signTypeIndex >= signTypeValues.size) {
-                    throw IllegalStateException("Illegal sign! ${signClassification.sign}")
-                }
+
+                check(signTypeIndex < signTypeValues.size) { "Illegal sign! ${signClassification.sign}" }
 
                 val signType = signTypeValues[signTypeIndex]
 
-                val signNumber = if (signTypeIndex > SignType.SpeedLimitRamp.ordinal) {
-                    SignNumber.Unknown
+                if (signType.hasNumber) {
```

Comment:
Previously signs from the top of enum were treated as those with numbers. New solution is not perfectly elegant either but still seems much better to me. If someone has better ideas lets discuss

Reply:
it looks like in future we'll separate numbers by region. In different counties the same signs could have or have no numbers

Reply:
Mmm, do you have some specific examples in mind? 
I think signs will be different then, there is no reason to name signs with the same name if one contains number and another does not.